### ISSUE TITLE
Do not suggest replacing class string inside `class_alias` second argumernt

### ISSUE THREAD

**Issue by asgraf** [hammer-tools/php-hammer#3]
Example code from my `src/Network/Http/Client.php`
```php
<?php
declare(strict_types=1);

namespace App\Network\Http;

use Cake\Core\Configure;

if (Configure::read('debug') && class_exists('\DebugHttp\Network\Http\Client')) {
    class_alias(\DebugHttp\Network\Http\Client::class, '\App\Network\Http\Client');
} else {
    class_alias(\Cake\Http\Client::class, '\App\Network\Http\Client');
}
```
This code triggers [Class name as string can be replaced](https://github.com/hammer-tools/php-hammer/wiki/Inspections#class-name-as-string-can-be-replaced) inspection
Since second argument of `class_alias` always should be some non-existent class name i don't think that replacing `'\App\Network\Http\Client'` with `\App\Network\Http\Client::class` is a good idea.
Although (to my surprise) such notation works correctly in PHP, I find it unintuitive and potentially confusing because it is reading a class name from a class that does not yet exist (we're just in the process of defining that alias). Doing so also triggers quality tools like phpstan 
That's why i suggest adding option to not suggest replacing class string inside `class_alias` second argument.

**Comment by rentalhost** [comment]
For PHP, the class notation is converted to a string naturally even if it doesn't exist (https://3v4l.org/p49Qb#v8.2.1), so using `class_alias()` or `class_exists()` with that notation is not a problem.

Personally, I don't know what would be ideal. However, I can add an option for it to be ignored in cases like this.

Could you say if there are other functions where this rule is applicable, in addition to those mentioned by you?

**Comment by asgraf** [comment]
I can't think of any cases when you refer to a non-existent class other than in `class_exists` and `class_alias`.

But as an second separate option you can limit conversion to a string ::class only if the class contained in the string exists in opened project.

**Comment by rentalhost** [comment]
So, in short, an "ignore rule if class doesn't exist" option solves all the problems, right?

**Comment by asgraf** [comment]
`ignore rule if class doesn't exist` rule does not take into account classes that are part of external dependencies that we want to support if they are included but we do not want to hard require them (for example CakePHP plugins)

So i think that there should be two separate options:
- first one for `class_exists` and `class_alias` cases
- and second one `ignore rule if class doesn't exist`
Both of them combined should cover all use cases

**Comment by rentalhost** [comment]
**Implemented to version 0.19.0:** it takes about three days to be published on the market. But if you want, you can download it from **[Releases](https://github.com/hammer-tools/php-hammer/releases/tag/0.19.0)**.

**Related changes:**

* Added an option to include functions like `class_exists()`, `class_alias()` and `interface_exists()` (_enabled_ by default) -- you may want to _disable_ it;
* Added an option to include non-existent classes to the analysis (_disabled_ by default) -- this was already the default, but now it is possible to activate it;

![image](https://user-images.githubusercontent.com/421851/214432328-caf4b894-db4e-4f7a-9c43-a1ae01df7893.png)
